src/test/java/codeu/controller/ChatServletTest.java

```diff
@@ -211,4 +214,52 @@ public void testDoPost_CleansHtmlContent() throws IOException, ServletException
 
     Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
   }
+
+  @Test
+  public void testDoPost_makePublicPrivate() throws IOException, ServletException {
+    Mockito.when(mockRequest.getRequestURI()).thenReturn("/chat/test_conversation");
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(
+        new User(
+          UUID.randomUUID(),
+          "test_username",
+          "testHash",
+          Instant.now()));
+    Mockito.when(mockConversationStore.getConversationWithTitle("test_conversation"))
+        .thenReturn(mockConversation);
+
+    Mockito.when(mockRequest.getParameter("type"))
+        .thenReturn("private");
+    Mockito.when(mockConversation.getType()).thenReturn("public");
+    
+    chatServlet.doPost(mockRequest, mockResponse);
+
+    Mockito.verify(mockConversation).makePrivate();
+    Mockito.verify(mockConversationStore).updateConversation(mockConversation);
+    Mockito.verify(mockResponse).sendRedirect("/chat/test_conversation");
+  }
+
+  @Test
+  public void testDoPost_makePrivatePrivate() throws IOException, ServletException {
```

Comment:
The name of this test can be a little confusing because the test makes sure a private isn't made private.

Reply:
lol I was blanking on names, all the others I could think of were very long. Any suggestions?

Reply:
Actually nvm. I changed what the test is testing.